Add blog post for kubeletstats receiver metrics updates (#6974)
open-telemetry/opentelemetry.io: content/en/blog/2025/kubeletstats-receiver-metrics-deprecation.md

---
title:
  'Kubernetes CPU Metrics in the kubeletstats Receiver: Transition from

>   `.cpu.utilization` to `.cpu.usage`'

Author: Ah cool, good to know!

---

> title:
>   'Kubernetes CPU Metrics in the kubeletstats Receiver: Transition from
>   .cpu.utilization to .cpu.usage'

Reviewer: Wow, that's a long title! 🤷🏼‍♂️ Nit in formatting:

```suggestion
title: >
  Kubernetes CPU Metrics in the kubeletstats Receiver: Transition from
  .cpu.utilization to .cpu.usage
```

linkTitle: kubeletstats Receiver metrics update
author: >
  [Christos Markou](https://github.com/ChrsMark) (Elastic), [Tyler
  Helmuth](https://github.com/TylerHelmuth) (Honeycomb), [Dmitrii
  Anoshin](https://github.com/dmitryax) (Cisco/Splunk)
date: 2025-05-30
issue: https://github.com/open-telemetry/opentelemetry.io/issues/6847
cSpell:ignore: Anoshin Dmitrii Helmuth Kubelet kubeletstats Kubelet’s Markou
---

The OpenTelemetry Collector’s

> [`kubeletstats`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.127.0/receiver/kubeletstatsreceiver/README.md)

Reviewer: Is there a reason why the URL here refers to v0.127.0, but v0.126.0 is used elsewhere in the page?

Reviewer: Btw, thanks for pinning those URLs to a specific version. That will help the links stay valid as the blog post ages. 🙌🏻

Author: I think at the time of writing the latest version was `v0.126.0` and after few days (this week) `v0.127.0` came out, so that's why we ended up with 2 versions most probably :). Thank's for spotting this!

Author: Right. That should be fixed now.

---
title:
  'Kubernetes CPU Metrics in the kubeletstats Receiver: Transition from
  .cpu.utilization to .cpu.usage'
linkTitle: kubeletstats Receiver metrics update
author: >
  [Christos Markou](https://github.com/ChrsMark) (Elastic), [Tyler
  Helmuth](https://github.com/TylerHelmuth) (Honeycomb), [Dmitrii
  Anoshin](https://github.com/dmitryax) (Cisco/Splunk)

> date: 2025-05-26

Reviewer 2: ```suggestion
date: 2025-05-30
```

issue: https://github.com/open-telemetry/opentelemetry.io/issues/6847
cSpell:ignore: Anoshin Dmitrii Helmuth Kubelet kubeletstats Kubelet’s Markou
---

> ## Kubernetes CPU Metrics in the `kubeletstats` Receiver: Transition from `.cpu.utilization` to `.cpu.usage`

Reviewer 2: No need to repeat the title here. Hugo automatically adds the front matter title as the page title:

<img width="762" alt="Screenshot 2025-05-29 at 3 58 39 PM" src="https://github.com/user-attachments/assets/3080c5e9-8de0-40f2-9649-b2921ace8b47" />

```suggestion

```

receiver emitted CPU metrics labeled with `.cpu.utilization`, such as:

- [`k8s.node.cpu.utilization`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.126.0/receiver/kubeletstatsreceiver/documentation.md#k8snodecpuutilization)
- [`k8s.pod.cpu.utilization`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.126.0/receiver/kubeletstatsreceiver/documentation.md#k8spodcpuutilization)
- [`container.cpu.utilization`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.126.0/receiver/kubeletstatsreceiver/documentation.md#containercpuutilization)

These metrics actually represent
[**raw CPU usage in cores**](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.126.0/receiver/kubeletstatsreceiver/internal/kubelet/cpu.go#L25-L26),
derived from the Kubernetes Kubelet’s
[`UsageNanoCores`](https://github.com/kubernetes/kubernetes/blob/8adc0f041b8e7ad1d30e29cc59c6ae7a15e19828/staging/src/k8s.io/kubelet/pkg/apis/stats/v1alpha1/types.go#L230-L233)
field, which is an absolute measure of CPU usage (in units of nanocores).

> The term _utilization_ generally implies a relative metric—a ratio or percentage

Reviewer 3: is `metric-a` a typo?
